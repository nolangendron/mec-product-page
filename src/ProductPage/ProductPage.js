import React, { Fragment } from 'react'
import { Header, SearchDetails, ProductsList, Pagination } from '../components'

export const ProductPage = ({
  searchQuery,
  handleQueryChange,
  currentSearch,
  handleSearch,
  products,
  currentProducts,
  paginate,
  paginateArrow,
  currentPage,
  productsPerPage,
}) => {
  return (
    <Fragment>
      <Header
        searchQuery={searchQuery}
        handleQueryChange={handleQueryChange}
        handleSearch={handleSearch}
      />
      <SearchDetails
        currentSearch={currentSearch}
        numberOfProducts={products.length}
      />
      <ProductsList products={currentProducts} />
      {products.length > 0 && (
        <Pagination
          currentPage={currentPage}
          paginateArrow={paginateArrow}
          paginate={paginate}
          productPerPage={productsPerPage}
          totalProducts={products.length}
        />
      )}
    </Fragment>
  )
}
