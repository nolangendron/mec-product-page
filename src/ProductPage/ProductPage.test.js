import React from 'react'
import { render } from '@testing-library/react'
import { ProductPage } from './ProductPage'

it('should take a snapshot of LandingPage', () => {
  const { asFragment } = render(<ProductPage />)
  expect(asFragment(<ProductPage />)).toMatchSnapshot()
})
