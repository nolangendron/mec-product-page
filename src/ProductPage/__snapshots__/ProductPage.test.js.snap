// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of LandingPage 1`] = `
<DocumentFragment>
  <header
    class="css-1u8elbe"
  >
    <img
      alt="MEC logo"
      class="flex-item logo"
      src="mec-logo.svg.png"
    />
    <div
      class="css-1lo72dc"
    >
      <form>
        <div
          class="MuiFormControl-root MuiTextField-root SearchInput-root-1"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root SearchInput-label-2 MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="standard-basic"
            id="standard-basic-label"
          >
            I'm looking for
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="standard-basic"
              type="text"
              value=""
            />
          </div>
        </div>
      </form>
    </div>
  </header>
  <div
    class="css-xf7pps"
  />
  <section
    class="css-5vjz5q"
  />
</DocumentFragment>
`;
